modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java

```diff
@@ -174,6 +190,47 @@ private void enrichParameter(OfValueInformation ofValueInformation,
                                        name, getValueProviderId(ofValueInformation.getValue())));
   }
 
+  private void enrichParameterFields(List<FieldValues> fieldsValues, ParameterDeclaration paramDeclaration,
+                                     Map<String, String> parameterNames, String name, List<ParameterDeclaration> allParameters) {
+
+    List<FieldValueProviderModel> fieldValueProviderModels = new LinkedList<>();
+    Map<String, ValueProviderFactoryModelProperty> valueProviderFactoryModelProperties = new HashMap<>();
+
+    for (FieldValues fieldValues : fieldsValues) {
+      ValueProviderFactoryModelPropertyBuilder propertyBuilder =
+          ValueProviderFactoryModelProperty.builder(fieldValues.value());
+
+      ParameterizableTypeWrapper resolverClassWrapper =
+          new ParameterizableTypeWrapper(fieldValues.value(), new DefaultExtensionsTypeLoaderFactory().createTypeLoader());
+      List<ExtensionParameter> resolverParameters = resolverClassWrapper.getParametersAnnotatedWith(Parameter.class);
+
+      resolverParameters.forEach(param -> propertyBuilder
+          .withInjectableParameter(param.getName(), param.getType().asMetadataType(), param.isRequired()));
+
+      Reference<Boolean> requiresConfiguration = new Reference<>(false);
+      Reference<Boolean> requiresConnection = new Reference<>(false);
+
+      enrichWithConnection(propertyBuilder, resolverClassWrapper)
+          .ifPresent(field -> requiresConnection.set(true));
+      enrichWithConfiguration(propertyBuilder, resolverClassWrapper)
+          .ifPresent(field -> requiresConfiguration.set(true));
+
+      int partOrder = 1;
+      for (String targetPath : fieldValues.targetPaths()) {
+        ValueProviderFactoryModelProperty valueProviderFactoryModelProperty = propertyBuilder.build();
+        valueProviderFactoryModelProperties.put(targetPath, valueProviderFactoryModelProperty);
+        fieldValueProviderModels
+            .add(new FieldValueProviderModel(getActingParametersModel(resolverParameters, parameterNames, allParameters),
+                                             requiresConfiguration.get(), requiresConnection.get(), fieldValues.open(),
+                                             partOrder,
+                                             name, getValueProviderId(fieldValues.value()), targetPath));
+        partOrder++;
+      }
+      paramDeclaration.setFieldValueProviderModels(fieldValueProviderModels);
```

Comment:
Add a validation in org.mule.runtime.module.extension.internal.loader.validation.ValueProviderModelValidator that validates that a parameter model can have value for the whole parameter, or values for fields of it, but not both.